### PR TITLE
[Merged by Bors] - chore(ContinuousMap): move `compRight*`

### PR DESCRIPTION
**Pull request by urkud**
Also drop some unneeded assumptions and rename/deprecate some lemmas.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by PatrickMassot** [comment]
Are those new names with precomp and postcomp part of a larger refactor? I like those names but I don’t think I ever saw anything like this in Mathlib so I fear this introduces inconsistencies.

**Comment by urkud** [comment]
I copied the names from [ContinuousLinearMap.precomp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Module/StrongTopology.html#ContinuousLinearMap.precomp)

File: Mathlib/Topology/ContinuousMap/Compact.lean [hunk 1]
```diff
@@ -417,57 +417,6 @@ end CompLeft
...
--/
-def compRightHomeomorph {X Y : Type*} (T : Type*) [TopologicalSpace X] [CompactSpace X]
-    [TopologicalSpace Y] [CompactSpace Y] [PseudoMetricSpace T] (f : X ≃ₜ Y) :
-    C(Y, T) ≃ₜ C(X, T) where
-  toFun := compRightContinuousMap T f
-  invFun := compRightContinuousMap T f.symm
-  left_inv g := ext fun _ => congr_arg g (f.apply_symm_apply _)
-  right_inv g := ext fun _ => congr_arg g (f.symm_apply_apply _)
-
-theorem compRightAlgHom_continuous {X Y : Type*} (R A : Type*) [TopologicalSpace X]
```
Comment (sgouezel) [inline Mathlib/Topology/ContinuousMap/Compact.lean:463]: Could you deprecate this name?

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Topology/ContinuousMap/Compact.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Topology/ContinuousMap/Compact.lean:463]: The renaming was accidental. I don't want to push neither way in `*_property` vs `property_*`. Renamed back.

**Comment by urkud** [comment]
Nothing on the queue, so let me try to merge it.
bors merge

**Pull request closed without merging**